localstack/services/secretsmanager/provider.py

```diff
@@ -0,0 +1,357 @@
+    ) -> ReplicateSecretToRegionsResponse:
+        return ReplicateSecretToRegionsResponse(**self._call_moto_with_request_secret_id(context))
+
+    def restore_secret(
+        self, context: RequestContext, secret_id: SecretIdType
+    ) -> RestoreSecretResponse:
+        return RestoreSecretResponse(**self._call_moto_with_request_secret_id(context))
+
+    def rotate_secret(
+        self,
+        context: RequestContext,
+        secret_id: SecretIdType,
+        client_request_token: ClientRequestTokenType = None,
+        rotation_lambda_arn: RotationLambdaARNType = None,
+        rotation_rules: RotationRulesType = None,
+        rotate_immediately: BooleanType = None,
+    ) -> RotateSecretResponse:
+        return RotateSecretResponse(**self._call_moto_with_request_secret_id(context))
+
+    def stop_replication_to_replica(
+        self, context: RequestContext, secret_id: SecretIdType
+    ) -> StopReplicationToReplicaResponse:
+        return StopReplicationToReplicaResponse(**self._call_moto_with_request_secret_id(context))
+
+    def tag_resource(
+        self, context: RequestContext, secret_id: SecretIdType, tags: TagListType
+    ) -> None:
+        self._call_moto_with_request_secret_id(context)
+
+    def untag_resource(
+        self, context: RequestContext, secret_id: SecretIdType, tag_keys: TagKeyListType
+    ) -> None:
+        self._call_moto_with_request_secret_id(context)
+
+    def update_secret(
+        self,
+        context: RequestContext,
+        secret_id: SecretIdType,
+        client_request_token: ClientRequestTokenType = None,
+        description: DescriptionType = None,
+        kms_key_id: KmsKeyIdType = None,
+        secret_binary: SecretBinaryType = None,
+        secret_string: SecretStringType = None,
+    ) -> UpdateSecretResponse:
+        return UpdateSecretResponse(**self._call_moto_with_request_secret_id(context))
+
+    def update_secret_version_stage(
+        self,
+        context: RequestContext,
+        secret_id: SecretIdType,
+        version_stage: SecretVersionStageType,
+        remove_from_version_id: SecretVersionIdType = None,
+        move_to_version_id: SecretVersionIdType = None,
+    ) -> UpdateSecretVersionStageResponse:
+        return UpdateSecretVersionStageResponse(**self._call_moto_with_request_secret_id(context))
+
+    def validate_resource_policy(
+        self,
+        context: RequestContext,
+        resource_policy: NonEmptyResourcePolicyType,
+        secret_id: SecretIdType = None,
+    ) -> ValidateResourcePolicyResponse:
+        return ValidateResourcePolicyResponse(**self._call_moto_with_request_secret_id(context))
+
+
+# maps key names to ARNs
+SECRET_ARN_STORAGE = {}
+
+
+def apply_patches():
+    def secretsmanager_models_secret_arn(region, secret_id):
+        k = "{}_{}".format(region, secret_id)
+        if k not in SECRET_ARN_STORAGE:
+            id_string = "".join(random.choice(string.ascii_letters) for _ in range(6))
```

Comment:
could be slightly simplified: `id_string = "".join(random.choices(string.ascii_letters, k=6))`
or maybe better yet `id_string = short_uid()[:6]` if it has to be exactly 6 characters long